website/static/js/oauthAddonNodeConfig.js

```diff
@@ -0,0 +1,245 @@
+            this.activateShare();
+        }
+    },
+    fetchEmailList: function(){
+        var self = this;
+
+        var ret = $.Deferred();
+        var promise = ret.promise();
+        if(!self.loadedEmails()){
+            $.ajax({
+                url: self.urls().emails,
+                type: 'GET',
+                dataType: 'json'
+            }).done(function(res){
+                self.loadedEmails(true);
+                ret.resolve(res.results.emails);
+            }).fail(function(xhr, status, error){
+                Raven.captureMessage('Could not GET ' + self.addonName + ' email list', {
+                    url: self.urls().emails,
+                    textStatus: status,
+                    error: error
+                });
+                ret.reject(xhr, status, error);
+            });
+        }
+        else{
+            ret.resolve(self.emails());
+        }
+        return promise;
+    },
+    activateShare: function() {
+        var self = this;
+        self.fetchEmailList()
+            .done(function(emails){
+                self.emails(emails);
+            });
+        var $copyBtn = $(self.selector).find('.copyBtn');
+        new ZeroClipboard($copyBtn);
+    },
+    _updateCustomFields: function(settings){
+        this.validCredentials(settings.validCredentials);
+    },
+     /**
+     * Allows a user to create an access token from the nodeSettings page
+     */
+    connectAccount: function() {
+        this.options.connectAccount.call(this);
+    },
+    importAuth: function(){
+        var self = this;
+
+        self.updateAccounts(function() {
+            if (self.accounts().length > 1) {
+                bootbox.prompt({
+                    title: 'Choose ' + self.addonName + ' Access Token to Import',
+                    inputType: 'select',
+                    inputOptions: ko.utils.arrayMap(
+                        self.accounts(),
+                        function(item) {
+                            return {
+                                text: item.name,
+                                value: item.id
+                            };
+                        }
+                    ),
+                    value: self.accounts()[0].id,
+                    callback: (self.connectExistingAccount.bind(self))
+                });
+            } else {
+                bootbox.confirm({
+                    title: 'Import ' + self.addonName + ' Access Token?',
+                    message: self.messages.confirmAuth(),
+                    callback: function(confirmed) {
+                        if (confirmed) {
+                            self.connectExistingAccount.call(self, (self.accounts()[0].id));
+                        }
+                    }
+                });
+            }
+        });
+    }
+});
+
+OauthAddonFolderPickerViewModel.prototype.connectExistingAccount = function(account_id) {
+    var self = this;
+
+    $osf.putJSON(
```

Comment:
Ideally, any method that does async work should return a promise. In this case. You can just return the return value of `$osf.putJSON`.